napari/_qt/qt_dims.py

```diff
@@ -328,3 +329,128 @@ def focus_down(self):
         else:
             index = (displayed.index(self.last_used) - 1) % len(displayed)
             self.last_used = displayed[index]
+
+    def play_dim(self, axis: int = 0, fps: float = 10, range=None):
+        """
+        Animate (play) axis
+
+        Parameters
+        ----------
+        axis: int
+            Index of axis to play
+        fps: float
+            Frames per second for playback (not guaranteed)
+        range: tuple | list
+            If specified, will constrain animation to [first, last] frames
+
+        Raises
+        ----------
+        IndexError
+            If ``axis`` requested is out of the range of the dims
+        IndexError
+            If ``range`` is provided and out of the range of the dims
+        ValueError
+            If ``range`` is provided and range[0] >= range[1]
+        """
+        # TODO: No access in the GUI yet. Just keybinding.
+        if axis >= len(self.dims.range):
+            raise IndexError('axis argument out of range')
+
+        dimsrange = self.dims.range[axis]
+        if range is not None:
+            if range[0] >= range[1]:
+                raise ValueError("animation range min must be <= range max")
+            if range[0] < dimsrange[0]:
+                raise IndexError("animation range min out of range")
+            if range[1] * dimsrange[2] >= dimsrange[1]:
+                raise IndexError("animation range max out of range")
+            self._animation_range = range
+        else:
+            self._animation_range = None
+        # allow only one axis to be playing at a time
+        # if nothing is playing self.stop() will not do anything
+        self.stop()
+
+        point = self.dims.point[axis]
+        self.animation_thread = AnimationThread(point + 1, axis, fps)
+        # when the thread timer increments, update the current frame
+        self.animation_thread.incremented.connect(self._set_frame)
+        self._set_frame(axis, point + 1)
+        self.animation_thread.start()
+
+    def stop(self):
+        """Stop axis animation"""
+        if self.is_playing:
+            self.animation_thread.quit()
+            self.animation_thread.wait()
+            del self.animation_thread
+            self.enable_play()
+
+    @property
+    def is_playing(self):
+        """Returns True if any axis is currently animated"""
+        return (
+            hasattr(self, 'animation_thread')
+            # this is repetive, since we delete the thread each time, but safer
+            and self.animation_thread.isRunning()
+        )
+
+    def _set_frame(self, axis, frame):
```

Comment:
I love how this is working, but one interaction pattern I'm curious if you have played with is what happens when you are playing an axis and you click in the slider for that axis. It looks like right now the animation actually stops. If you click again then it seems to start. The ideal interactivity would be if you just jumped to the frame where the mouse was clicked and then you continued to animate. Do you think this is possible?

Reply:
yeah that should be possible.  lemme play with it.

Reply:
see https://github.com/napari/napari/pull/607#issuecomment-549182306  It doesn't jump straight to the frame... but it doesn't do that even when the animation is not playing.  So if we want to jump to the frame, we should fix that elsewhere. but if we do, this will also have that behavior.